packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/endpoint_client/endpoint_global_fields_manager.dart

```diff
@@ -17,15 +18,43 @@ import 'package:amplify_secure_storage_dart/amplify_secure_storage_dart.dart';
 /// {@endtemplate}
 class EndpointGlobalFieldsManager {
   /// {@macro amplify_analytics_pinpoint_dart.endpoint_global_fields_manager}
+  @visibleForTesting
   EndpointGlobalFieldsManager(
     this._endpointInfoStore,
-    this._globalAttributes,
-    this._globalMetrics,
   );
 
+  /// {@macro amplify_analytics_pinpoint_dart.endpoint_global_fields_manager}
+  static Future<EndpointGlobalFieldsManager> create(
+    SecureStorageInterface endpointInfoStore,
+  ) async {
+    final fieldsManager = EndpointGlobalFieldsManager(endpointInfoStore);
+    await fieldsManager.init();
+    return fieldsManager;
+  }
+
+  /// {@macro amplify_analytics_pinpoint_dart.endpoint_global_fields_manager}
+  @visibleForTesting
+  Future<void> init() async {
+    /// Retrieve stored GlobalAttributes
+    final cachedAttributes =
+        await _endpointInfoStore.read(key: _endpointGlobalAttrsKey);
```

Comment:
Can these keys be tracked via `EndpointKeys`?

Reply:
As in an enum?  Sure updated